### PR TITLE
Add elevation_gain metric for components

### PR DESCRIPTION
**Pull request by lgangitano**
In search for a tool helping evaluate braking components wear, I've added elevation_gain reporting as a proxy for the amount of braking over time.

Rationale for this approach is that more elevation gain will, most of the time, imply also elevation loss and thus braking.

Further enhancement can be implemented extracting more data from Strava splits, as detailed here: https://groups.google.com/g/strava-api/c/rBYiuBFNzUE

### CODE REVIEW

File: src/strava_gear/data.py [hunk 1]
```diff
@@ -164,14 +166,16 @@ def all_rule_bike_ids(self) -> Set[BikeId]:
 @dataclass
 class Usage:
     distances: Dict[ComponentId, float] = field(default_factory=lambda: defaultdict(float))
+    elevation_gain: Dict[ComponentId, float] = field(default_factory=lambda: defaultdict(float))
```
Comment (liskin) [inline src/strava_gear/data.py:169]: This one should probably be `elevation_gains` for consistency.

**Review by liskin — COMMENTED** [review]
Looks like a sensible thing to do indeed, thanks for the contribution. 🙂

I'm mostly staying away from computers these days for mental health reasons, so apologies for being slow to react (now and possibly going forward as well).

File: src/strava_gear/data.py (hunk 1 above)
Comment (lgangitano) [reply src/strava_gear/data.py]: Thank you, I've fixed it.

**Comment by lgangitano** [comment]
Thanks for taking time to review it. If you think you're going to merge it, I can update the documentation as well, add the new column to the example.

**Comment by liskin** [comment]
When rebasing on top of current main, I made some small changes:
* no column renaming, so it's called `total_elevation_gain` in CSV input as well
* renamed the report column to "vert m" to make it clearer it's elev gain
* added `--show-vert-m / --hide-vert-m` defaulting to hide to reduce tests changes and because I believe most people won't need it (I can be convinced otherwise if you feel strongly about this)

Thanks a lot for the contribution!

**Comment by lgangitano** [comment]
Thanks!!!

**Comment by liskin** [comment]
Just a heads up that `--show-vert-m` was renamed to `--show-vert` in https://github.com/liskin/strava-gear/pull/11 which adds support for imperial metrics.

**Pull request merged**